play/src/components/joingame/SelectAvatar.tsx

```diff
@@ -0,0 +1,185 @@
+  // container for monster icons in grid
+  height: '118px',
+  width: '98px',
+  display: 'flex',
+  justifyContent: 'center',
+  alignItems: 'center',
+  borderRadius: '20px',
+});
+
+interface IconProps {
+  isSelected: boolean;
+}
+
+const Icon = styled('img', {
+  shouldForwardProp: (prop) => prop !== 'isSelected',
+})<IconProps>(({ isSelected }) => ({
+  height: '106px',
+  width: 'auto',
+  boxShadow: '0px 8px 20px rgba(26, 100, 136, 0.3)',
+  borderRadius: '20px',
+  borderColor: 'white',
+  borderStyle: 'solid',
+  animation: isSelected ? `scaleAnimation 300ms` : 'none',
+  '@keyframes scaleAnimation': {
+    '0%': {
+      opacity: 0,
+      transform: 'scale(1)',
+    },
+    '50%': {
+      opacity: 1,
+      transform: 'scale(0.95)',
+    },
+    '100%': {
+      opacity: 1,
+      transform: 'scale(1.0)',
+    },
+  },
+})
+);
+
+const MonsterContainer = styled(Box)({
+  display: 'flex',
+  justifyContent: 'center',
+  alignItems: 'flex-end',
+  padding: 0,
+});
+
+const Monster = styled('img')({
+  width: '250px',
+  height: 'auto',
+  animation: `none`,
+  '@keyframes bounceAnimation': {
+    '0%': {
+      opacity: 0,
+      transform: 'scale(1)',
+    },
+    '50%': {
+      opacity: 1,
+      transform: 'scale(0.95)',
+    },
+    '100%': {
+      opacity: 1,
+      transform: 'scale(1.0)',
+    },
+  },
+});
+
+const BottomContainer = styled(Box)({
+  display: 'flex',
+  flexDirection: 'column',
+  alignItems: 'center',
+  gap: 12,
+});
+
+interface SelectAvatarProps {
+  selectedAvatar: number | null;
+  handleAvatarSelected: (value: number) => void;
+  playerFirstName: string;
+  playerLastName: string;
+  isMobileDevice: boolean;
+}
+
+export default function SelectAvatar({
+  selectedAvatar,
+  handleAvatarSelected,
+  playerFirstName,
+  playerLastName,
+  isMobileDevice,
+}: SelectAvatarProps) {
+  const theme = useTheme();
+
+  return (
+    <JoinGameBackgroundContainer>
+      <StackContainer>
+        <Stack spacing={2}>
+          <Typography
+            variant="h2"
+            sx={{
+              textAlign: 'center',
+              paddingTop: `${theme.sizing.mediumPadding}px`,
+            }}
+          >
+            Choose Your Avatar!
+          </Typography>
+          <GridContainer>
+            {Object.keys(monsterMap).map((value, index) => (
+              <IconContainer key={uuidv4()}>
+                <Icon
+                  src={monsterMap[index].icon}
+                  onClick={() => {
+                    handleAvatarSelected(index);
+                  }}
+                  isSelected = {index === selectedAvatar}
+                  alt="avatar"
+                  sx={{ borderWidth: index === selectedAvatar ? '6px' : '0px' }}
+                />
+              </IconContainer>
+            ))}
+          </GridContainer>
+        </Stack>
+        <MonsterContainer sx={{ height: isMobileDevice ? '250px' : '300px' }}>
+          <Monster
+            src={monsterMap[selectedAvatar || 0].monster}
```

Comment:
Since we're defaulting it to `0` here, should this property just not be nullable and has a default value of `0`?

Reply:
I'm trying to double up the use of selectedAvatar here (for both the monster image and the monster icon) but there's an edge case this is catching. When a player first navigates to the screen, selectedAvatar is null because I don't think we want any selection on the avatar image. However, the main monster image should still be at `0` so that a default is selected (we ran into crashes with the `mobile` app when we didn't do this). The condition is below:

![image](https://user-images.githubusercontent.com/79417944/234687990-f8dcc5f5-83d4-4334-92d6-6cfd1d2b3fdd.png)

This is in contrast to when a player selects the first monster. In that case selectedAvatar goes to 0, and the white border does appear:

![image](https://user-images.githubusercontent.com/79417944/234688371-c792387c-0120-4d41-95c3-3c0ef6d7ae9f.png)

So I'd like to maintain a default of null (to prevent the white border from appearing) while also providing a default value here so a monster always shows. I think this is working intended but it's definitely not super clear. I've added a comment that explains this line in the code and if you have another way of improving the readability, I'm all ears.

Reply:
I think there should be a correlation between monster photos and the main one shown below. If we wanna go with `null`, then probably we shouldn't also show the large monster below. It's a bit confusing if initial load has no monsters selected, but shows the first monster image. Especially, if we use `null`, we should disable `choose` button so user cannot move on to the next screen without choosing an avatar.

Reply:
Yeah, touche. I'm off tomorrow and Friday but let me talk to Yong Lin about how the flow should work so we can produce a better correlation. I'll follow up with a PR next week.